Update xPSSessionConfiguration to fix PSSA and style issues

Dan1el42 opened the pull request:
1. Fixed PSSA and styles issues
2. Renamed internal functions to follow verb-noun format
3. Decorated all functions with comment-based help
4. Updated Unreleased section of README.md

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/powershell/xpsdesiredstateconfiguration/208)
<!-- Reviewable:end -->

FayzaSultan commented on DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1:
```diff
@@ -60,154 +87,285 @@ function Get-TargetResource
 
+<#
+    .SYNOPSIS
+        Ensures the specified PSSessionConfiguration is its desired state
```
> is its desired state [](start = 52, length = 21)

is in its desired state.

Dan1el42 replied:
Thanks very much for catching this. I've added the missing word in two lines and updated the PR.

kwirkykat commented:
All the $params variables need to have more descriptive names.

---

Reviewed 1 of 2 files at r1.
Review status: 1 of 2 files reviewed at latest revision, 3 unresolved discussions.

---

*[DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1, line 68 \[r1\]](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208#-KQqgVqBqs-xAASSFHCv:-KQqgVqBqs-xAASSFHCw:b3nfenh) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/e9c18e912cab34d4052c81e9d25053844b31eae8/DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1#L68)):*
> ```PowerShell
>         if ($endpoint.RunAsUser)
>         {
>             $params = @{
> ```

$params needs a more descriptive name

---

*[DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1, line 169 \[r1\]](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208#-KQqh0k6gH6Rwx8_yqof:-KQqh0k6gH6Rwx8_yqog:bvdaj48) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/e9c18e912cab34d4052c81e9d25053844b31eae8/DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1#L169)):*
> ```PowerShell
> 
>     if ($PSCmdlet.ShouldProcess(($LocalizedData.EnsureSessionConfigurationMessage -f $Ensure)))
>     {
> ```

Everything inside this if statement needs to be indented once more

---

*[DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1, line 619 \[r1\]](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208#-KQqhY83Ja6ldS49mpwr:-KQqhY87b6UvtnMpPjS4:b-1vf470) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/e9c18e912cab34d4052c81e9d25053844b31eae8/DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1#L619)):*
> ```PowerShell
>     if ($Apply)
>     {
>         $Parameters = @{}
> ```

$Parameters should be capitalized (and it should have a more descriptive name, but you didn't write the variable name so it's ok)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208#-:-KQqibWd_g_5ywDCJvLL:bco5k8q)*
<!-- Sent from Reviewable.io -->

Dan1el42 commented:
Review status: 1 of 2 files reviewed at latest revision, 3 unresolved discussions.

---

*[DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1, line 169 \[r1\]](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208#-KQqh0k6gH6Rwx8_yqof:-KQs_GlUg12UUFSBdoKS:b-319eal) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/e9c18e912cab34d4052c81e9d25053844b31eae8/DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1#L169)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Everything inside this if statement needs to be indented once more
</details>
The indentation looks alright to me in VS Code and ISE. Please have another look.

![xPSSessionConfiguration\_line\_169.PNG](https://files.reviewableusercontent.io/images/reviewable/7493263/ACshR_XxrfFjkzfuv2vvb8e-/xPSSessionConfiguration_line_169.png)

---

*[DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1, line 619 \[r1\]](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208#-KQqhY83Ja6ldS49mpwr:-KQsgfQ_UkOT73iBR-vf:bo74a6t) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/e9c18e912cab34d4052c81e9d25053844b31eae8/DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1#L619)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> $Parameters should be capitalized (and it should have a more descriptive name, but you didn't write the variable name so it's ok)
</details>
If updated the variable name to be more descriptive as suggested.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208)*
<!-- Sent from Reviewable.io -->

Dan1el42 commented:
PR updated as per your comments. Please resume the review at your earliest convenience.

---

Review status: 1 of 2 files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208#-:-KQshUJ2gu4UM5DLyP7U:brivkyw)*
<!-- Sent from Reviewable.io -->

kwirkykat commented:
Review status: 1 of 2 files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---

*[DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1, line 169 \[r1\]](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208#-KQqh0k6gH6Rwx8_yqof:-KR09odwp_q-xuZlYxsx:baqtjp9) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/e9c18e912cab34d4052c81e9d25053844b31eae8/DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1#L169)):*
<details><summary><i>Previously, Dan1el42 (Daniel Krebs) wrote…</i></summary>
> The indentation looks alright to me in VS Code and ISE. Please have another look.
> 
> ![xPSSessionConfiguration\_line\_169.PNG](https://files.reviewableusercontent.io/images/reviewable/7493263/ACshR_XxrfFjkzfuv2vvb8e-/xPSSessionConfiguration_line_169.png)
> 
</details>
Looks fine now :smile: Not sure what I was seeing before...

---

*[DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1, line 181 \[r2\]](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208#-KR0ATATb8M6DbnX0cpd:-KR0ATATb8M6DbnX0cpe:bp06dkh) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/e0f36a8df398020b4641f46b5a6e578d4f09857d/DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1#L181)):*
> ```PowerShell
>                 {
>                     # Set the following preference so the functions inside Unregister-PSSessionConfig
>                     # doesn't get these settings
> ```

Please convert this multi-line function to the <# ... #> format

---

*[DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1, line 238 \[r2\]](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208#-KR0C9Sli7Jp81C9_hbc:-KR0C9Sli7Jp81C9_hbd:blu3bl1) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/e0f36a8df398020b4641f46b5a6e578d4f09857d/DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1#L238)):*
> $setPSSessionConfigurationParams = $validatedProperties.psobject.Copy()
> $setPSSessionConfigurationParams['Force'] = $true
> $setPSSessionConfigurationParams['NoServiceRestart'] = $true
> $setPSSessionConfigurationParams['Verbose'] = $false
>  $null = Set-PSSessionConfiguration @setPSSessionConfigurationParams

Why not just  ```$null = Set-PSSessionConfiguration @validatedProperties -Force -NoRestartService -Verbose:$false``` ?

---

*[DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1, line 267 \[r2\]](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208#-KR0CaUyBJny8ozW2l0Y:-KR0CaUyBJny8ozW2l0Z:bkt95dc) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/e0f36a8df398020b4641f46b5a6e578d4f09857d/DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1#L267)):*
> @('Ensure','Verbose','Debug')
Can you please add a space after each comma in this list?

---

*[DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1, line 279 \[r2\]](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208#-KR0CmCgRk6Iz54BLLdj:-KR0CmCgRk6Iz54BLLdk:bp06dkh) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/e0f36a8df398020b4641f46b5a6e578d4f09857d/DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1#L279)):*
> ```PowerShell
>                 {
>                     # Set the following preference so the functions inside
>                     # Unregister-PSSessionConfig doesn't get these settings
> ```

Please convert this multi-line function to the <# ... #> format

---

*[DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1, line 306 \[r2\]](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208#-KR0EBlY8Msq8KHoW3cK:-KR0EBlY8Msq8KHoW3cL:b-vvtfkc) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/e0f36a8df398020b4641f46b5a6e578d4f09857d/DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1#L306)):*
> $parameters/$validatedProperties
I can't find where this variable is initialized within the else in the old code or in the updated code.
Am I missing it somewhere or is this a bug that needs to get filed?

---

*[DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1, line 369 \[r2\]](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208#-KR0EzJZqqgeF6l0xR0c:-KR0EzJZqqgeF6l0xR0d:bc7v791) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/e0f36a8df398020b4641f46b5a6e578d4f09857d/DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1#L369)):*
>         To ensure that it does, set this property to "Present".
>         To ensure that it does not exist, set the property to "Absent".

To test that it exists, set this pro...
To test that it does not exist, set this pro...

---

*[DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1, line 540 \[r2\]](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208#-KR0G6nRq9dLKUfzXGp6:-KR0G6nRq9dLKUfzXGp7:b-eh8sqi) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/e0f36a8df398020b4641f46b5a6e578d4f09857d/DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1#L540)):*
> Parameters
Please change the parameter name in the help comment as it was changed in the function

---

*[DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1, line 553 \[r2\]](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208#-KR0Fx1MmnhpMHPVQrqf:-KR0Fx1MmnhpMHPVQrqg:bfm2b6n) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/e0f36a8df398020b4641f46b5a6e578d4f09857d/DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1#L553)):*
>  $validatedProperties,
Please capitalize this variable since it is a parameter.

---

*[DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1, line 599 \[r2\]](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208#-KR0GKEERBHqqSoLD1xt:-KR0GKEERBHqqSoLD1xu:b-dghrli) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/e0f36a8df398020b4641f46b5a6e578d4f09857d/DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1#L599)):*
> Get-ResourcePropertyTable
Can you please rename this function to Get-ValidatedResourcePropertyTable since it also validates all the properties?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208)*
<!-- Sent from Reviewable.io -->

Dan1el42 commented:
Review status: 1 of 2 files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---

*[DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1, line 238 \[r2\]](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208#-KR0C9Sli7Jp81C9_hbc:-KR1HdwhgneqzNGGvXBq:bog3ihk) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/e0f36a8df398020b4641f46b5a6e578d4f09857d/DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1#L238)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> > $setPSSessionConfigurationParams = $validatedProperties.psobject.Copy()
> > $setPSSessionConfigurationParams['Force'] = $true
> > $setPSSessionConfigurationParams['NoServiceRestart'] = $true
> > $setPSSessionConfigurationParams['Verbose'] = $false
> >  $null = Set-PSSessionConfiguration @setPSSessionConfigurationParams
> 
> Why not just  ```$null = Set-PSSessionConfiguration @validatedProperties -Force -NoRestartService -Verbose:$false``` ?
</details>
The line exceeded our 100 character limit by 21 characters.

---

*[DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1, line 306 \[r2\]](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208#-KR0EBlY8Msq8KHoW3cK:-KR1IRvcrRFboccJlqX0:bcua2tg) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/e0f36a8df398020b4641f46b5a6e578d4f09857d/DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1#L306)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> > $parameters/$validatedProperties
> I can't find where this variable is initialized within the else in the old code or in the updated code.
> Am I missing it somewhere or is this a bug that needs to get filed?
</details>
The variable is being initialized in line 227 of revision 2 or line 229 of revision 3.

---

*[DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1, line 540 \[r2\]](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208#-KR0G6nRq9dLKUfzXGp6:-KR1JJJqs5mg-XIbe6qX:bsta672) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/e0f36a8df398020b4641f46b5a6e578d4f09857d/DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1#L540)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> > Parameters
> Please change the parameter name in the help comment as it was changed in the function
</details>
Ups. My mistake. I've changed it back to Parameters for this function.

---

*[DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1, line 553 \[r2\]](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208#-KR0Fx1MmnhpMHPVQrqf:-KR1JcO3UumWxkiJVWXF:b-8zgtoz) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/e0f36a8df398020b4641f46b5a6e578d4f09857d/DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1#L553)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> >  $validatedProperties,
> Please capitalize this variable since it is a parameter.
</details>
Changed back to Parameters. Mistake on my part.

---

*[DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1, line 599 \[r2\]](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208#-KR0GKEERBHqqSoLD1xt:-KR1JsD_jlMZNYauThiF:bxbk7dv) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/e0f36a8df398020b4641f46b5a6e578d4f09857d/DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1#L599)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> > Get-ResourcePropertyTable
> Can you please rename this function to Get-ValidatedResourcePropertyTable since it also validates all the properties?
</details>
Function rename. Good catch.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208)*
<!-- Sent from Reviewable.io -->

kwirkykat commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> Please resolve merge conflicts by rebasing per instructions [here](https://github.com/PowerShell/DscResources/blob/master/GettingStartedWithGitHub.md#resolve-merge-conflicts).

---

Reviewed 1 of 2 files at r3.
Review status: 1 of 2 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1, line 306 \[r2\]](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208#-KR0EBlY8Msq8KHoW3cK:-KR5Otwu4qvIkgCB1F2V:bv9q6gb) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/e0f36a8df398020b4641f46b5a6e578d4f09857d/DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1#L306)):*
<details><summary><i>Previously, Dan1el42 (Daniel Krebs) wrote…</i></summary>
> The variable is being initialized in line 227 of revision 2 or line 229 of revision 3.
</details>
That initialization is in a different logical path. I don't think this variable ever gets initialized if endpoint is $null/$false/0.
I can file a separate bug for this if you would like since it looks like this was a problem in the resource before this PR.

```
if ($endpoint)
{
    if ($Ensure -eq 'Absent')
    {
        ...
    }
    else
    {
        $parameters / $validatedProperties initialized
        $parameters / $validatedProperties used
    }
}
else
{
    $parameters / $validatedProperties used <--- we are here
}

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208#-:-KR5fQb4q5rhGAg4eUxR:bcpcrib)*
<!-- Sent from Reviewable.io -->

Dan1el42 commented:
@kwirkykat I've been on the move since Tuesday without access to suitable computer. I'll resolve the merge conflict on Wednesday September 14th. Thanks for your understanding.

Dan1el42 commented:
Review status: 1 of 2 files reviewed at latest revision, all discussions resolved.

---

*[DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1, line 306 at r2](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208#-KR0EBlY8Msq8KHoW3cK:-KRe36kVouSZEaVu3ZHj:b9x1757) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/e0f36a8df398020b4641f46b5a6e578d4f09857d/DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1#L306)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> That initialization is in a different logical path. I don't think this variable ever gets initialized if endpoint is $null/$false/0.
> I can file a separate bug for this if you would like since it looks like this was a problem in the resource before this PR.
> 
> ```
> if ($endpoint)
> {
>     if ($Ensure -eq 'Absent')
>     {
>         ...
>     }
>     else
>     {
>         $parameters / $validatedProperties initialized
>         $parameters / $validatedProperties used
>     }
> }
> else
> {
>     $parameters / $validatedProperties used <--- we are here
> }
> 
</details>
You're right. I think we can safely remove below because as you've highlighted $parameters/$validatedProperties is only being initialized in the other logical path. I'm happy to remove it and update the PR.

```
                    # If the $validatedProperties contain more than 1 key, something needs to be changed
                    if ($validatedProperties.count -gt 1)
                    {
                        Write-EndpointMessage -Parameters $validatedProperties -keysToSkip 'Name'
                    }
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208)*
<!-- Sent from Reviewable.io -->

kwirkykat commented:
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1, line 306 at r2](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208#-KR0EBlY8Msq8KHoW3cK:-KRed6LIJSHGg4Qz0lBb:bxdo2c3) ([raw file](https://github.com/powershell/xpsdesiredstateconfiguration/blob/e0f36a8df398020b4641f46b5a6e578d4f09857d/DSCResources/MSFT_xPSSessionConfiguration/MSFT_xPSSessionConfiguration.psm1#L306)):*
<details><summary><i>Previously, Dan1el42 (Daniel Krebs) wrote…</i></summary>
> You're right. I think we can safely remove below because as you've highlighted $parameters/$validatedProperties is only being initialized in the other logical path. I'm happy to remove it and update the PR.
> 
> ```
>                     # If the $validatedProperties contain more than 1 key, something needs to be changed
>                     if ($validatedProperties.count -gt 1)
>                     {
>                         Write-EndpointMessage -Parameters $validatedProperties -keysToSkip 'Name'
>                     }
> ```
</details>
Sure. You can remove it since it will never get called anyway

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208)*
<!-- Sent from Reviewable.io -->

Dan1el42 commented:
@kwirkykat I've removed the inactive code as agreed. If you're happy with my recent modifications it would be great if we can get this PR merged into dev. I would like to rebase my branch of the composite resource PR #209 to resolve the merge conflict of the README.md. Thanks very much.

kwirkykat commented:
Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xpsdesiredstateconfiguration/208)*
<!-- Sent from Reviewable.io -->
